src/yarpWholeBodySensors.cpp

```diff
@@ -943,23 +1203,118 @@ bool yarpWholeBodySensors::readAccelerometer(const int accelerometer_index, doub
+            double *tempData;
+            tempData = mtbLastRead[accelerometer_runtime_index].data();
+            acc[0] = tempData[mtb_first_data_index] * (double)5.9855e-4;
+            acc[1] = tempData[mtb_first_data_index+1] * (double)5.9855e-4;
+            acc[2] = tempData[mtb_first_data_index+2] * (double)5.9855e-4;
+            ret = true;
+            break;
+        default :
+            ret = false;
+            break;
+    }
+    return ret;
+}
 
-        acc[0] = imuLastRead[accelerometer_imu_index][4];
-        acc[1] = imuLastRead[accelerometer_imu_index][5];
-        acc[2] = imuLastRead[accelerometer_imu_index][6];
-        ret = true;
+bool yarpWholeBodySensors::readGyroscope(const int gyroscope_index, double *gyro, double *stamps, bool wait)
+{
+    bool ret = true;
+    if( gyroscope_index >= (int)sensorIdList[wbi::SENSOR_GYROSCOPE].size() || gyroscope_index < 0 )
+    {
+        return false;
     }
-    else
+    int gyroscope_runtime_index;
+    switch(gyroscopesReferenceIndeces[gyroscope_index].type)
     {
-        ret = false;
+        case IMU_GYRO :
+            gyroscope_runtime_index = gyroscopesReferenceIndeces[gyroscope_index].runtime_index;
+
+            // Process IMU accelerometer data
+            if( stamps != 0 )
+            {
+                *stamps = imuStampLastRead[gyroscope_runtime_index];
+            }
+
+            gyro[0] = imuLastRead[gyroscope_runtime_index][7];
+            gyro[1] = imuLastRead[gyroscope_runtime_index][8];
+            gyro[2] = imuLastRead[gyroscope_runtime_index][9];
+            ret = true;
+            break;
+
+        case MTB_GYRO :
+            // Process MTB gyroscope data
+            gyroscope_runtime_index = gyroscopesReferenceIndeces[gyroscope_index].runtime_index;
+            yarp::sig::Vector *vt;
+
+            vt = portsMTBSensors[gyroscope_runtime_index]->read(wait);
+           if(vt!=NULL) {
+                if(mtbLastRead[gyroscope_runtime_index].size()==0)
+                {
+                    mtbLastRead[gyroscope_runtime_index].resize(vt->size(),0.0);
+                }
+                mtbLastRead[gyroscope_runtime_index] = *vt;
+                yarp::os::Stamp info;
+                portsMTBSensors[gyroscope_runtime_index]->getEnvelope(info);
+                mtbStampLastRead[gyroscope_runtime_index] = info.getTime();
+            }
+            if( stamps != 0 )
+            {
+                *stamps = mtbStampLastRead[gyroscope_runtime_index];
+            }
+            int mtb_first_data_index;
+            mtb_first_data_index = gyroscopesReferenceIndeces[gyroscope_index].data_index-1;
+            double *tempData;
+            tempData = mtbLastRead[gyroscope_runtime_index].data();
+            gyro[0] = tempData[mtb_first_data_index] * (double)7.6274e-3;
+            gyro[1] = tempData[mtb_first_data_index+1] * (double)7.6274e-3;
+            gyro[2] = tempData[mtb_first_data_index+2] * (double)7.6274e-3;
```

Comment:
What is the unit of measure of this conversion factors? As far as I remember the iCub IMU is publishing the angular velocity with `deg/s`, we may make sure that also this is `deg/s` or switch all of them to `[rad]/s` for consistency, the important thing is that they are all consistent.